trollsift/parser.py

```diff
@@ -265,158 +285,165 @@ def parse(self, format_string):
             literal_text = self._escape(literal_text)
             yield literal_text, field_name, format_spec, conversion
 
-    def get_value(self, key, args, kwargs):
+    def get_value(
+        self, key: int | str, args: Sequence[Any], kwargs: Mapping[str, Any]
```

Comment:
Why do you sometimes use `Mapping` instead of `dict`?

Reply:
I suppose I could use `Mapping` everywhere I use `dict`, I honestly didn't even think about it.

Reply:
I could use MutableMapping for read/write. Yeah not sure if that's more clear or more confusing.

Reply:
Ok I updated it to use `Mapping`. It seems like everywhere I passed a `dict` argument it was read-only so seems reasonable.